www/public-interface.js

```diff
@@ -180,7 +180,14 @@ module.exports = function init(exec, cookieHandler, urlUtil, helpers, globalConf
         break;
       case 'upload':
         var fileOptions = helpers.checkUploadFileOptions(options.filePath, options.name);
-        exec(onSuccess, onFail, 'CordovaHttpPlugin', 'uploadFiles', [url, headers, fileOptions.filePaths, fileOptions.names, options.connectTimeout, options.readTimeout, options.followRedirect, options.responseType, reqId]);
+        var hasProgressCallback = options.onProgress != null;
```

Comment:
You need to check for falsey value here not for null, because it will be `undefined` if not provided.
```suggestion
        var hasProgressCallback = !!options.onProgress;
```